services/src/main/java/org/keycloak/services/Urls.java

```diff
@@ -273,4 +277,33 @@ private static UriBuilder themeBase(URI baseUri) {
     public static URI samlRequestEndpoint(final URI baseUri, final String realmName) {
         return realmBase(baseUri).path(RealmsResource.class, "getProtocol").build(realmName, SamlProtocol.LOGIN_PROTOCOL);
     }
+
+    public static void checkUrl(KeycloakSession session, String url, String name) throws IllegalArgumentException{
+        if (url == null) {
+            return;
+        }
+
+        URL parsed;
+
+        try {
+            parsed = new URL(url);
+        } catch (MalformedURLException e) {
+            throw new IllegalArgumentException("The url [" + name + "] is malformed", e);
+        }
+
+        String protocol = parsed.getProtocol().toLowerCase();
+
+        if (!protocol.startsWith("http")) {
+            throw new IllegalArgumentException("Invalid protocol/scheme for url [" + name + "]");
+        }
+
+        ClientConnection connection = session.getContext().getConnection();
+
+        if (connection != null) {
+            if (!"https".equals(protocol) && session.getContext().getRealm()
+                    .getSslRequired().isRequired(connection)) {
+                throw new IllegalArgumentException("The url [" + name + "] requires secure connections");
```

Comment:
This is not correct. You are checking here if SSL is required to the IP address of the admin that is adding the identity provider. It should be checking if SSL is required for the address of the identity provider.